update/update.go

```diff
@@ -574,127 +588,49 @@ func checkBaseBootstrapSwitch() {
-		"rquery", "-U", "%v", "os/userland",
-	)
-	err := cmd.Run()
+func updatercscript() {
+	// Intercept the /etc/rc script
+	src := defines.STAGEDIR + "/etc/rc"
+	dest := defines.STAGEDIR + "/etc/rc-updatergo"
+	cpCmd := exec.Command("mv", src, dest)
+	err := cpCmd.Run()
 	if err != nil {
-		return
-	}
-
-	// We have os/userland remote, lets see if we are using it already locally
-	cmd = exec.Command(defines.PKGBIN, "query", "%v", "os/userland")
-	err = cmd.Run()
-	if err == nil {
-		return
+		log.Fatal(err)
 	}
 
-	output, cmderr := exec.Command(
-		"tar", "czf", defines.STAGEDIR+"/.etcbackup.tgz", "-C",
-		defines.STAGEDIR+"/etc", ".",
-	).Output()
-	if cmderr != nil {
-		ws.SendMsg(string(output))
-		ws.SendMsg("Failed saving /etc configuration", "fatal")
+	var fuflag string
+	if defines.FullUpdateFlag {
+		fuflag = "-fullupdate"
 	}
-	// Make sure pkg is fetched
-	ws.SendMsg("Fetching new base")
-	logger.LogToFile("Fetching new base")
-	cmd = exec.Command(
-		defines.PKGBIN, "-C", defines.PkgConf, "fetch", "-U", "-d",
-		"-y", "os/userland", "os/kernel",
-	)
-	err = cmd.Run()
-	if err != nil {
-		ws.SendMsg("Failed fetching new base", "fatal")
+	var cacheflag string
+	if defines.CacheDirFlag != "" {
+		cacheflag = "-cachedir " + defines.CacheDirFlag
 	}
 
-	// Get list of packages we need to nuke
-	shellcmd := defines.PKGBIN +
-		" query '%o %n-%v' | grep '^base ' | awk '{print $2}'"
-	output, cmderr = exec.Command("/bin/sh", "-c", shellcmd).Output()
-	if cmderr != nil {
-		ws.SendMsg("Failed getting base package list", "fatal")
+	var upflag string
+	if defines.UpdateFileFlag != "" {
+		upflag = "-updatefile " + defines.UpdateFileFlag
 	}
 
-	basepkgs := strings.TrimSpace(string(output))
-	barr := strings.Split(basepkgs, "\n")
-	for i := range barr {
-		// Unset vital flag
-		ws.SendMsg("Removing: " + barr[i])
-		logger.LogToFile("Removing: " + barr[i])
-		cmd := exec.Command(
-			defines.PKGBIN, "-c", defines.STAGEDIR, "-C", defines.PkgConf,
-			"set", "-v", "0", barr[i],
-		)
-		err := cmd.Run()
-		if err != nil {
-			log.Fatal("Failed unsetting vital")
-		}
-		// Remove the package now
-		cmd = exec.Command(defines.PKGBIN, "-c", defines.STAGEDIR,
-			"-C", defines.PkgConf, "delete", "-y", "-f", "-g", barr[i],
-		)
-		remout, err := cmd.CombinedOutput()
-		if err != nil {
-			ws.SendMsg(string(remout))
-			ws.SendMsg("Failed removing "+barr[i], "fatal")
-		}
-		if strings.Contains(barr[i], "-runtime-1") {
-			// If this was the runtime package, need to re-install userland
-			// right away
-			ws.SendMsg("Boot-strapping userland")
-			pkgcmd := exec.Command(
-				defines.PKGBIN, "-r", defines.STAGEDIR, "-C", defines.PkgConf,
-				"install", "-U", "-f", "-y", "os/userland",
-			)
-			fullout, err := pkgcmd.CombinedOutput()
-			if err != nil {
-				ws.SendMsg(string(fullout))
-				ws.SendMsg("Failed boot-strapping userland", "fatal")
-			}
-		}
+	selfbin, _ := os.Executable()
+	ugobin := "/." + defines.ToolName
```

Comment:
Do you mean ./?

Reply:
No, this was correct. Hidden binary with '/.' just for style.